code/datums/mapgen/biomes/_biome.dm

```diff
@@ -2,39 +2,146 @@
 /datum/biome
 	///Type of turf this biome creates
 	var/turf_type
-	///Chance of having a structure from the flora types list spawn
+	/// Chance of having a structure from the flora types list spawn
 	var/flora_density = 0
-	///Chance of having a mob from the fauna types list spawn
+	/// Chance of spawning special features, such as geysers.
+	var/feature_density = 0
+	/// Chance of having a mob from the fauna types list spawn
 	var/fauna_density = 0
-	///list of type paths of objects that can be spawned when the turf spawns flora
-	var/list/flora_types = list(/obj/structure/flora/grass/jungle/a/style_random)
-	///list of type paths of mobs that can be spawned when the turf spawns fauna
+	/// Weighted list of type paths of flora that can be spawned when the
+	/// turf spawns flora.
+	var/list/flora_types = list()
+	/// Weighted list of extra features that can spawn in the biome, such as
+	/// geysers. Gets expanded automatically.
+	var/list/feature_types = list()
+	/// Weighted list of type paths of fauna that can be spawned when the
+	/// turf spawns fauna.
 	var/list/fauna_types = list()
 
+
+/datum/biome/New()
+	. = ..()
+	if(length(flora_types))
+		flora_types = expand_weights(fill_with_ones(flora_types))
+
+	if(length(fauna_types))
+		fauna_types = expand_weights(fill_with_ones(fauna_types))
+
+	if(length(feature_types))
+		feature_types = expand_weights(feature_types)
+
+
 ///This proc handles the creation of a turf of a specific biome type
 /datum/biome/proc/generate_turf(turf/gen_turf)
```

Comment:
similar question bout this does it exist to be overrided?

Reply:
Yeah, it exists to be overriden. It's for stuff like geysers, so they can be added through the biomes instead of the cavegenerator itself, so you could have a biome with slightly more geysers, or more chances of rarer geysers, or really, ***any*** other feature we might want.

Reply:
Oh, wait, `generate_turf()` was added by floyd when he added biomes. I didn't want to break the `jungle_generator`, so I left these procs as-is.